src/backend/distributed/utils/reference_table_utils.c

```diff
@@ -150,6 +152,20 @@ ReplicateAllReferenceTablesToNode(char *nodeName, int nodePort)
 		Oid referenceTableId = lfirst_oid(referenceTableCell);
 		List *shardIntervalList = LoadShardIntervalList(referenceTableId);
 		ShardInterval *shardInterval = (ShardInterval *) linitial(shardIntervalList);
+
+		referenceShardIntervalList = lappend(referenceShardIntervalList,
+											 shardInterval);
+	}
+
+	if (ClusterHasKnownMetadataWorkers())
+	{
+		BlockWritesToShardList(referenceShardIntervalList);
+	}
+
+	foreach(referenceShardIntervalCell, referenceShardIntervalList)
+	{
+		ShardInterval *shardInterval = (ShardInterval *) lfirst(
+			referenceShardIntervalCell);
 		uint64 shardId = shardInterval->shardId;
 
 		LockShardDistributionMetadata(shardId, ExclusiveLock);
```

Comment:
After talking with @onderkalaci we decided to keep the implementation like this, otherwise non-MX cluster won't acquire the required lock. Beside, we want to keep MX cluster check logic on this function instead of `BlockWritesToShardList`